JavaScript/JavaScriptSDK/TelemetryContext.ts

```diff
@@ -145,12 +161,35 @@ module Microsoft.ApplicationInsights {
 
             sessionStateEnvelope.time = Util.toISOStringForIE8(new Date(timestamp));
 
-            tc._track(sessionStateEnvelope); 
+            tc._track(sessionStateEnvelope);
+        }
+
+        private _applyCustomProperties(envelope: Microsoft.Telemetry.Envelope) {
+            if (this.properties && envelope && envelope.data && (<any>envelope.data).baseData) {
```

Comment:
Do we expect all of these conditions? Would it be more appropriate to [fail fast](http://martinfowler.com/ieeeSoftware/failFast.pdf) instead?